File: guides/micronaut-scope-types/java/src/main/java/example/micronaut/refreshable/Robot.java
package example.micronaut.refreshable;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.runtime.context.scope.Refreshable;
import java.util.UUID;

@Refreshable
public class Robot {
    @NonNull
    private final String serialNumber;

    public Robot() {
        serialNumber = UUID.randomUUID().toString();
    }

    @NonNull
    public String getSerialNumber() {
        return serialNumber;
    }
}

File: guides/micronaut-scope-types/kotlin/src/main/kotlin/example/micronaut/refreshable/Robot.kt
package example.micronaut.refreshable

import io.micronaut.runtime.context.scope.Refreshable
import java.util.UUID

@Refreshable // <1>
class Robot {
    private val serialNumber: String = UUID.randomUUID().toString()

    fun getSerialNumber(): String {
        return serialNumber
    }
}